perun-core/src/main/java/cz/metacentrum/perun/core/entry/GroupsManagerEntry.java

```diff
@@ -1458,4 +1458,39 @@ public boolean canExtendMembershipInGroupWithReason(PerunSession sess, Member me
 
 		return getGroupsManagerBl().getIndirectMembershipPaths(sess, member, group);
 	}
+
+	@Override
+	public List<RichMember> getGroupRichMembersByIds(PerunSession sess, int groupId, List<Integer> memberIds, List<String> attrNames) throws GroupNotExistsException, PrivilegeException, AttributeNotExistsException  {
+		Utils.checkPerunSession(sess);
+
+		Group group = groupsManagerBl.getGroupById(sess, groupId);
+		getGroupsManagerBl().checkGroupExists(sess, group);
+
+		// Authorization
+		if ( !AuthzResolver.authorizedInternal(sess, "getGroupRichMembersByIds_int_List<Integer>_List<String>_policy", group)) {
+			throw new PrivilegeException(sess, "getGroupRichMembersByIds");
+		}
+
+		List<AttributeDefinition> attributeDefinitions = new ArrayList<>();
+		for (String attrName : attrNames) {
+			attributeDefinitions.add(getPerunBl().getAttributesManagerBl().getAttributeDefinition(sess, attrName));
+		}
+
+		List<Member> members = perunBl.getMembersManagerBl().getMembersByIds(sess, memberIds);
```

Comment:
I'm afraid this won't return the group statuses of the members, which is the main reason why we need this new method. 

If you want you can use the `getGroupMemberById` method. This method returns the searched member with all of its group statuses. 

If you want to return all of the members by a single SQL into the DB, you would need to create a new method on the bl and impl layer as well. You could modify the already present method `getGroupMembersById` so it would take a list of ids and use the SQL `IN` clause. 

The choice is yours.

Reply:
Thanks, I opted for the first choice.